src/mediapipe_internal/pythonnoderesource.cpp

```diff
@@ -75,14 +81,30 @@ Status PythonNodeResource::createPythonNodeResource(std::shared_ptr<PythonNodeRe
     std::string parentPath = fsHandlerPath.parent_path();
     std::string filename = fsHandlerPath.filename();
 
+    // node_name validated in graph definition
+    std::string node_name = graphNode.name();
+
     py::gil_scoped_acquire acquire;
     try {
+        py::list inputStreams = py::list();
+        py::list outputStreams = py::list();
+        for (auto& name : graphNode.input_stream()) {
+            inputStreams.append(name);
+        }
+
+        for (auto& name : graphNode.output_stream()) {
+            outputStreams.append(name);
+        }
+
         py::module_ sys = py::module_::import("sys");
         sys.attr("path").attr("append")(parentPath.c_str());
         py::module_ script = py::module_::import(filename.c_str());
         py::object OvmsPythonModel = script.attr("OvmsPythonModel");
         py::object pythonModel = OvmsPythonModel();
         py::object kwargsParam = pybind11::dict();
+        kwargsParam["input_stream"] = inputStreams;
+        kwargsParam["output_stream"] = outputStreams;
+        kwargsParam["node_name"] = node_name;
         py::bool_ success = pythonModel.attr("initialize")(kwargsParam);
```

Comment:
Can you rework this part as well? We agreed not to return anything from initialize.